OptimoveSDK/Sources/Classes/Optimobile/AnalyticsHelper.swift

```diff
@@ -163,6 +163,12 @@ final class AnalyticsHelper {
     }
 
     private func syncEvents(context: NSManagedObjectContext?, _ onSyncComplete: SyncCompletedBlock? = nil) {
+        self.serialQueue.async {
+            self.syncEventsImpl(context: self.analyticsContext, onSyncComplete)
+        }
+    }
+
+    private func syncEventsImpl(context: NSManagedObjectContext?, _ onSyncComplete: SyncCompletedBlock? = nil) {
         context?.performAndWait {
```

Comment:
Another thing to improve my general understanding/knowledge - since we're using `performAndWait`, could this cause threading issues or deadlocks? Would `perform` work?

Reply:
Although wasn't changed, it is very important to have  `performAndWait` it is done on the current thread (serial queue), guaranteeing that the block will finish before next one starts